src/platform_impl/windows/event_loop.rs

```diff
@@ -2148,5 +2184,14 @@ unsafe extern "system" fn thread_event_target_callback<T: 'static>(
         _ => commctrl::DefSubclassProc(window, msg, wparam, lparam),
     };
 
-    runner.catch_unwind(callback).unwrap_or(-1)
+    let result = subclass_input
+        .event_loop_runner
+        .catch_unwind(callback)
+        .unwrap_or(-1);
+    if ncdestroy {
+        remove_event_target_window_subclass(window, subclass_input);
```

Comment:
I'm not really happy with that solution tbh - unfortunately the topic is quite complex to get right due to the recursive call of the window proc (e.g DestroyWindow will invoke the callback again for th NCDESTROY case etc) and this approach makes it harder to maintain imo.
To be fair I currently don't have a clear opinion on the best way either. I played a bit around with local dereferencing but it doesn't feel robust either: https://github.com/msiglreith/winit/commit/9fd2ceea6763f4db60d52ecb62d2c85daaff6d49

Reply:
Would returning `(LRESULT, Option<Box<SubclassInput<T>>>)` from the `callback` closure and calling `Box::into_raw` in the `(_, Some(_))` case be better?

Reply:
**Or** (assuming that I've understood the recursive nature of `WndProc` correctly): The `subclass_input: SubclassInput<T>` could be changed to be `Rc<SubclassInput<T>>` which always has one "fake" reference. At the beginning of each call to `*_callback` we'd deref the `subclass_input_ptr` and clone the `Rc`, which would ensure that we never drop the `subclass_input` before all calls to the callback are done. `WM_NCDESTROY` would then deregister the subclass and remove the "fake" reference, in that order.